src/common/transformations/src/transformations/op_conversions/einsum_decomposition.cpp

```diff
@@ -395,6 +400,211 @@ void reduce_input(ngraph::pass::EinsumDecomposition *einsum_decompose_ptr,
+        new_shape.insert(new_shape.end(), separate_shape.begin(), separate_shape.end());
+        new_shape.insert(new_shape.end(), reduced_shape.begin(), reduced_shape.end());
+    } else {
+        new_shape.insert(new_shape.end(), reduced_shape.begin(), reduced_shape.end());
+        new_shape.insert(new_shape.end(), separate_shape.begin(), separate_shape.end());
+    }
+
+    const ngraph::Shape old_shape = input.get_shape();
+    if (old_shape == new_shape) {
+        return;
+    }
+    const auto old_shape_size = old_shape.size();
+    const auto new_shape_size = new_shape.size();
+    NGRAPH_CHECK(old_shape_size <= new_shape_size);
+
+    const auto new_shape_const = ngraph::opset7::Constant::create(ngraph::element::Type_t::i64, ngraph::Shape {new_shape.size()}, new_shape);
+    const auto broadcast = std::make_shared<ngraph::opset7::Broadcast>(input, new_shape_const, ngraph::op::BroadcastType::NUMPY);
+
+    inputs[input_ind] = broadcast->output(0);
+
+    subgraph_nodes.insert(subgraph_nodes.end(), {new_shape_const, broadcast});
+}
+
+ngraph::Output<ngraph::Node> build_identity(const ngraph::Output<ngraph::Node>& input_node,
+                                            const std::vector<size_t>& repeated_label_dims,
+                                            ngraph::NodeVector& subgraph_nodes) {
+    NGRAPH_CHECK(repeated_label_dims.size() > 1);
+    const auto input_shape = input_node.get_shape();
+    ngraph::Shape identity_shape(input_shape.size(), 1);
+    const size_t repeated_label_dim_size = input_shape[repeated_label_dims[0]];
+    for (const auto dim : repeated_label_dims) {
+        NGRAPH_CHECK(dim < input_shape.size());
+        NGRAPH_CHECK(repeated_label_dim_size == input_shape[dim]);
+        identity_shape[dim] = repeated_label_dim_size;
+    }
+
+    const size_t number_of_identity_elements = shape_size(identity_shape);
+    std::vector<int> identity_values(number_of_identity_elements, 0);
+    const size_t p = repeated_label_dim_size;
+    if (p == 1) {
+        identity_values[0] = 1;
+    } else {
+        const size_t n = repeated_label_dims.size();
+        const size_t alpha = (static_cast<size_t>(std::pow(p, n)) - 1) / (p - 1);
+        size_t offset = 0;
+        for (size_t k = 0; k < p; ++k) {
+            identity_values[offset] = 1;
+            offset += alpha;
+        }
+    }
+    const auto identity = ngraph::opset7::Constant::create(input_node.get_element_type(), identity_shape, identity_values);
+    subgraph_nodes.insert(subgraph_nodes.end(), {identity});
+    return subgraph_nodes.back();
+}
+
+ngraph::Output<ngraph::Node> build_multi_identity(ngraph::pass::EinsumDecomposition* einsum_decompose_ptr,
+                                                  const ngraph::Output<ngraph::Node>& input_node,
+                                                  const std::vector<std::string>& repeated_labels,
+                                                  LabelDimMap& label_dim_map,
```

Comment:
I think it is a copy-paste from reference part I have written. But the problem with such approach that the transformation is not reshapable. The transformation makes a model non-reshapable.
We can't accept this implementation

Reply:
We agreed with @p-durandin  to make implementation inside transformations.
If "reshapable" means to provide dynamic shape support - it was not expected in the scope of this PR. Could you clarify a little bit about needed changes?

Reply:
@tgubanova-lohika, @p-durandin, sorry but I does not support to break reshapability of this transformations that is used by different plugins. It is not a good idea. You can create a separate transformation for GPU plugin and execute it there.